### ISSUE TITLE
Query variable not returning all of the expected values

### ISSUE THREAD

**Issue by snwbr** [grafana/grafana#13843]
Read before posting: 

Please include this information:
### What Grafana version are you using?
v5.3.2

### What datasource are you using?
Prometheus

### What OS are you running grafana on?
CentOS Linux 7

### What did you do?
I've created a query variable. When performing the query, not all of the results are shown, even though Prometheus is showing them all. If Grafana's GET action is replayed, intermittently all of the results appear. But it never does it the first time. I'm testing with a low ammount of results (34) but it only return 32 or 33, sometimes the 34 values.

### What was the expected result?
All of the results queried from Prometheus should appear on the GET action.

### What happened instead?
Some of the results are missing.

### Include raw network request & response: get by opening Chrome Dev Tools (F12, Ctrl+Shift+I on windows, Cmd+Opt+I on Mac), go the network tab.

GET /api/datasources/proxy/1/api/v1/series?match[]=probe_http_version&start=1540571733&end=1540571793 HTTP/1.1
Host: 10.251.172.54
Connection: keep-alive
Accept: application/json, text/plain, */*
X-Grafana-Org-Id: 1
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36
Referer: http://XX.XX.XX.XX/d/-P-XxhKiz/services-health-dev?editview=templating&refresh=30s&orgId=1&panelId=&fullscreen=&edit=
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9,es;q=0.8
Cookie: grafana_user=admin; grafana_remember=f26b45e5c5c6e81e269d667befd08487abe322ec9a875224e23d6355d2c691812b; grafana_sess=aa4abc131abcd321

HTTP/1.1 200 OK
Access-Control-Allow-Headers: Accept, Authorization, Content-Type, Origin
Access-Control-Allow-Methods: GET, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: Date
Cache-Control: no-cache
Content-Encoding: gzip
Content-Length: 381
Content-Type: application/json
Date: Fri, 26 Oct 2018 16:36:32 GMT
Expires: -1
Pragma: no-cache

(Changed the endpoints for security reasons), but it illustrates that endpoint 33 is missing).
{"status":"success","data":[
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/1/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/2/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/3/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/4/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/5/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/6/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/7/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/8/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/9/hcheck","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/10/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/11/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/12/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/13/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/14/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/15/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/16/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/17/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/18/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/19/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/20/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/21/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/22/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/23/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/24/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/25/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/26/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/27/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/28/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/29/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/30/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/31/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/32/health","job":"blackbox"},
{"__name__":"probe_http_version","instance":"https://dev-foo.bar.com/34/health","job":"blackbox"}]}

**Comment by snwbr** [comment]
Please don’t close this one. This is not a Prometheus issue, if you pay attention to what I wrote you can see that it’s the /api/datasources endpoint in Grafana  what’s not responding accordingly. If I query Prometheus directly it returns all of the data right away.

**Comment by snwbr** [comment]
As I said, I've tried the prometheus api and it returns all the data all the times, no errors there. But some results are still disappearing in Grafana. So I keep my version that this is not something in Prometheus. Perhaps it may be the api wrapper in Grafana.

**Comment by torkelo** [comment]
Strange, we have no other reports of the proxy causing issues like this. When you call the prometheus api directly are you also adding the identical (changing) time range to the query? (&start=1540571733&end=1540571793) which will change for every refersh.

**Comment by snwbr** [comment]
Yes, I am. Actually, I just got to a workaround for me: originally, I was using label_values(probe_http_version,instance) in the Query field for creating a query variable, that would eventually call the Grafana proxy api and the rest is history.

But when I changed to use query_result(probe_http_version) instead, all of the values are returned always, no issue at all.

**Comment by snwbr** [comment]
I've played a bit with the time stamps and saw this:

Got the current timestamp from https://www.unixtimestamp.com/ and use that value as `start` field, added a hundred of seconds in the `end` parameter. When running the Grafana api, 10-15 results are gotten from a total of 35. At the same time, uses the same parameters for Prometheus endpoint, returning all the 35 values.

That may explain why if I keep refreshing Grafana I'll eventually get all the data, since the seconds pass and the query gets to a point in which is able to collect all the results as expected. If I change the query in grafana for a custom range, it works fine.

**Comment by torkelo** [comment]
So the exact same call returned different number of items? How is that possible, grafana just returns what Prometheus returns. Super strange

**Comment by snwbr** [comment]
I know. Do you have by any chance a Prometheus instance somewhere to replicate this?

**Comment by Ehekatl** [comment]
Might just run into this as well, not on query variables but graphs.

Query over prom always return full data (30 success out of 30 tries), but query on grafana randomly missing data for hours (10~15 success out of 30 tries), query inspector show the query return 200 status, but the data points are missing.

Any idea how to debugging this?

**Comment by jtaleric** [comment]
I am hitting this. I can query prom directly and get the desired result, however through Grafana I am missing values. I will see if I can get the Chrome `dev-tools` to show the request/response -- it would be a great feature to add the query inspector to the variables page too...
